src/Propel/Generator/Model/Table.php

```diff
@@ -33,200 +36,100 @@ class Table extends ScopedMappingModel implements IdMethod
-
-    /**
-     * @var string
-     */
-    private $commonName;
-
-    /**
-     * @var string
-     */
-    private $originCommonName;
-
-    /**
-     * @var string
-     */
-    private $description;
-
-    /**
-     * @var string
-     */
-    private $phpName;
-
-    /**
-     * @var string
-     */
-    private $idMethod;
-
-    /**
-     * @var bool
-     */
-    private $allowPkInsert = false;
-
-    /**
-     * @var string|null
-     */
-    private $phpNamingMethod;
+    private array $idMethodParameters = [];
 
-    /**
-     * @var \Propel\Generator\Model\Database|null
-     */
-    private $database;
+    private ?string $commonName = null;
+    private string $originCommonName;
+    private ?string $description = null;
+    private ?string $phpName = null;
+    private string $idMethod;
+    private bool $allowPkInsert = false;
+    private ?string $phpNamingMethod = null;
+    private ?Database $database = null;
 
     /**
      * @var array<\Propel\Generator\Model\ForeignKey>
      */
-    private $referrers = [];
-
-    /**
-     * @var bool
-     */
-    private $containsForeignPK = false;
-
-    /**
-     * @var \Propel\Generator\Model\Column|null
-     */
-    private $inheritanceColumn;
-
-    /**
-     * @var bool
-     */
-    private $skipSql = false;
-
-    /**
-     * @var bool
-     */
-    private $readOnly = false;
-
-    /**
-     * @var bool
-     */
-    private $isAbstract = false;
-
-    /**
-     * @var string|null
-     */
-    private $alias;
+    private array $referrers = [];
 
-    /**
-     * @var string|null
-     */
-    private $interface;
-
-    /**
-     * @var string|null
-     */
-    private $baseClass;
+    private bool $containsForeignPK = false;
+    private ?Column $inheritanceColumn = null;
+    private bool $skipSql = false;
+    private bool $readOnly = false;
+    private bool $isAbstract = false;
+    private ?string $alias = null;
+    private ?string $interface = null;
+    private ?string $baseClass = null;
+    private ?string $baseQueryClass = null;
 
     /**
-     * @var string|null
+     * @var string[]
      */
-    private $baseQueryClass;
+    private array $columnsByName = [];
 
     /**
-     * @var array
-     */
-    private $columnsByName = [];
-
-    /**
-     * @var array
+     * @var string[]
      */
     private $columnsByLowercaseName = [];
 
     /**
-     * @var array
+     * @var string[]
      */
     private $columnsByPhpName = [];
 
-    /**
-     * @var bool
-     */
-    private $needsTransactionInPostgres = false;
-
-    /**
-     * @var bool
-     */
-    private $heavyIndexing = false;
-
-    /**
-     * @var bool|null
-     */
-    private $identifierQuoting;
-
-    /**
-     * @var bool|null
-     */
-    private $forReferenceOnly;
-
-    /**
-     * @var bool
-     */
-    private $reloadOnInsert = false;
-
-    /**
-     * @var bool
-     */
-    private $reloadOnUpdate = false;
+    private bool $needsTransactionInPostgres = false;
```

Comment:
There are other prs that address types adding, let's please keep this minimal to the change at Hand.
Also for easier Review.